submissions/management/commands/update_submissions_uuids.py

```diff
@@ -56,12 +57,12 @@ def handle(self, *args, **options):
         total_len = Submission.objects.count()
         log.info("Beginning uuid update, {} rows exist in total".format(total_len))
 
-        current = options['start'];
+        current = options['start']
         while current < total_len:
             end_chunk = current + options['chunk'] if total_len - options['chunk'] >= current else total_len
             log.info("Updating entries in range [{}, {})".format(current, end_chunk))
             with transaction.atomic():
-                for submission in Submission.objects.filter(id__gte=current, id__lt=end_chunk).iterator():
+                for submission in Submission.objects.all()[current:end_chunk].iterator():
```

Comment:
I'm going to verify this for sure, but I'm reasonably positive that it goes like this:

- `all()` maps to a `SELECT (*)`. Query is still unevaluated.
- slice is applied to our queryset. Since it's still unevaluated, the ORM is smart and appends the appropriate `LIMIT` statements to our query, which continues to not be evaluated.
- `iterator()` actually performs the query, one row at a time (using more limits?)

I'll grab the raw SQL from a debugging session and get back to you.